cloudprem/app/scaling.tf

```
@@ -0,0 +1,87 @@
+data "kubernetes_all_namespaces" "allns" {
+  depends_on = [helm_release.replicated]
+}
+
+resource "kubernetes_horizontal_pod_autoscaler" "app" {
+  depends_on = [helm_release.replicated]
+  metadata {
+    name = "app-hpa"
+    # Terraform magic required to find the randomly created replicated namespace so we can install the HPAs in the right place.
+    namespace = coalesce([for i, v in data.kubernetes_all_namespaces.allns.namespaces : try(regexall("replicated\\-.*", v)[0], "")]...)
+  }
+
+  spec {
+    min_replicas = 2
+    max_replicas = 50
```

I'd recommend starting with a lower max here. Otherwise a big scale out could swap the whole cluster.
```suggestion
    max_replicas = 10
```

---

The storage tier wont scale elastically though? So we need to be careful where we set the max thresholds.